src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotUseInsecureSettingsForJsonNetTests.cs

```diff
@@ -856,25 +857,31 @@ public class MyISerializationBinder : Newtonsoft.Json.Serialization.ISerializati
         [Fact]
         public async Task Secure_FieldInitialization_BinderSet_NoDiagnostic()
```

Comment:
After changing AdditionalMetadataReferences.DefaultWithNewtonsoftJson to use version 12.0.3, the test can be:

```csharp
        [Fact]
        public async Task Secure_FieldInitialization_BinderSet_NoDiagnostic()
        {
            await VerifyCSharpWithJsonNetAsync(@"
using System;
using Newtonsoft.Json;

class Blah
{
    public static readonly JsonSerializerSettings Settings = 
        new JsonSerializerSettings()
        {
            TypeNameHandling = TypeNameHandling.Objects,
            Binder = new MyBinder(),
        };
}

public class MyBinder : System.Runtime.Serialization.SerializationBinder
{
    public override Type BindToType(string assemblyName, string typeName) => throw new NotImplementedException();
    public override void BindToName(Type serializedType, out string assemblyName, out string typeName) => throw new NotImplementedException();
}");
        }
```

Reply:
The tests have been updated to test against both 10.0.1 and 12.0.1.